app/models/vm_or_template/operations.rb

```diff
@@ -112,11 +112,12 @@ def validate_vm_control
             end
       unsupported_reason_add(:control, msg) if msg
     end
-  end
 
-  def validate_terminate
-    return {:available => false, :message => 'The VM is terminated'} if self.terminated?
-    {:available => true, :message => nil}
+    supports :terminate do
+      msg = unsupported_reason(:control) unless supports_control?
+      msg ||= _("Provider doesn't support vm_destroy") unless ext_management_system.respond_to?(:vm_destroy)
```

Comment:
when checking for two reason we just added all reasons
```ruby
unsupported_reason_add(:vm_destroy, unsupported_reason(:control)) unless supports_control?
unsupported_reason_add(:vm_destroy, _("Provider doesn't support vm_destroy") unless ext_management_system.respond_to?(:vm_destroy))
```

This way we could later even stack the reasons when asking for this.